File: packages/android/src/main/java/com/kozmos/components/ToggleButton/ToggleButton.figma.kt
package com.kozmos.components.togglebutton

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=383-2019")
class KozmosToggleButtonConnect {
    @FigmaProperty(FigmaType.Text, "Label Text")
    val label: String = "Toggle"

    @FigmaProperty(FigmaType.Enum, "Variant")
    val variant: KozmosToggleButtonVariant = Figma.mapping(
        "Default" to KozmosToggleButtonVariant.Default,
        "Outline" to KozmosToggleButtonVariant.Outline
    )

    @FigmaProperty(FigmaType.Enum, "Size")
    val size: KozmosToggleButtonSize = Figma.mapping(
        "Small" to KozmosToggleButtonSize.Small,
        "Default" to KozmosToggleButtonSize.Default,
        "Large" to KozmosToggleButtonSize.Large
    )

    @FigmaProperty(FigmaType.Enum, "State")
    val checked: Boolean = Figma.mapping(
        "Default" to false,
        "Pressed" to true,
        "Disabled" to false,
        "Focus" to false
    )

    @Composable
    fun ComponentExample() {
        KozmosToggleButton(
            checked = checked,
            onCheckedChange = {},
            label = label,
            variant = variant,
            size = size
        )
    }
}

File: packages/ios/Sources/Components/ToggleButton/ToggleButton.figma.swift
import SwiftUI
import Figma

struct KozmosToggleButtonConnect: FigmaConnect {
    let component = KozmosToggleButton.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=383-2019"

    @FigmaString("Label Text")
    var label: String = "Toggle"

    @FigmaEnum(
        "Variant",
        mapping: [
            "Default": "default",
            "Outline": "outline"
        ]
    )
    var variantName: String = "default"

    @FigmaEnum(
        "Size",
        mapping: [
            "Small": "sm",
            "Default": "default",
            "Large": "lg"
        ]
    )
    var sizeName: String = "default"

    @FigmaEnum(
        "State",
        mapping: [
            "Default": false,
            "Pressed": true,
            "Disabled": false,
            "Focus": false
        ]
    )
    var isOn: Bool = false

    var body: some View {
        KozmosToggleButton(
            isOn: .constant(self.isOn),
            label: self.label,
            variant: self.variantName == "outline" ? .outline : .default,
            size: self.sizeName == "sm" ? .sm : (self.sizeName == "lg" ? .lg : .default)
        )
    }
}
